hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -15,8 +15,1580 @@
+                       x) ==
+                     boost::any_cast<
+                       boost::optional<std::vector<boost::optional<int32_t>>>>(
+                       y);
+          },
+          [](const boost::any& x, const boost::any& y) { //[41] NULLABLE_INT64
+              return boost::any_cast<boost::optional<int64_t>>(x) ==
+                     boost::any_cast<boost::optional<int64_t>>(y);
+          },
+          [](const boost::any& x,
+             const boost::any& y) { //[42] ARRAY_OF_NULLABLE_INT64
+              return boost::any_cast<
+                       boost::optional<std::vector<boost::optional<int64_t>>>>(
+                       x) ==
+                     boost::any_cast<
+                       boost::optional<std::vector<boost::optional<int64_t>>>>(
+                       y);
+          },
+          [](const boost::any& x, const boost::any& y) { //[43] NULLABLE_FLOAT32
+              return boost::any_cast<boost::optional<float>>(x) ==
+                     boost::any_cast<boost::optional<float>>(y);
+          },
+          [](const boost::any& x,
+             const boost::any& y) { //[44] ARRAY_OF_NULLABLE_FLOAT32
+              return boost::any_cast<
+                       boost::optional<std::vector<boost::optional<float>>>>(
+                       x) ==
+                     boost::any_cast<
+                       boost::optional<std::vector<boost::optional<float>>>>(y);
+          },
+          [](const boost::any& x, const boost::any& y) { //[45] NULLABLE_FLOAT64
+              return boost::any_cast<boost::optional<double>>(x) ==
+                     boost::any_cast<boost::optional<double>>(y);
+          },
+          [](const boost::any& x,
+             const boost::any& y) { //[46] ARRAY_OF_NULLABLE_FLOAT64
+              return boost::any_cast<
+                       boost::optional<std::vector<boost::optional<double>>>>(
+                       x) ==
+                     boost::any_cast<
+                       boost::optional<std::vector<boost::optional<double>>>>(
+                       y);
+          }
+      };
+
+    const pimpl::schema& xs = x.get_schema();
+    const pimpl::schema& ys = y.get_schema();
+
+    if (xs != ys)
+        return false;
+
+    for (const std::pair<std::string, boost::any>& xp : x.objects_) {
+        const std::string& field_name = xp.first;
+        const boost::any& value_of_x = xp.second;
+        const boost::any& value_of_y = y.objects_.at(field_name);
+
+        boost::optional<pimpl::field_descriptor> kind_opt =
+          xs.get_field(field_name);
+
+        if (!COMPARATORS[std::size_t(kind_opt->kind)](value_of_x, value_of_y))
+            return false;
+    }
+
+    return true;
+}
+
+bool
+operator!=(const generic_record& x, const generic_record& y)
+{
+    return !(x == y);
+}
+}
+
+std::ostream&
+operator<<(std::ostream& os, field_kind kind)
+{
+    switch (kind) {
+        case field_kind::BOOLEAN:
```

Comment:
enumeration value 'NOT_AVAILABLE' not handled in switch

Reply:
Fixed.